include/deal.II/lac/scalapack.h

```diff
@@ -118,9 +118,9 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
    * and distributed using the grid @p process_grid.
    */
   ScaLAPACKMatrix(
-    const size_type                                           n_rows,
-    const size_type                                           n_columns,
-    const std::shared_ptr<const Utilities::MPI::ProcessGrid> &process_grid,
+    const size_type                                          n_rows,
+    const size_type                                          n_columns,
+    const std::shared_ptr<const Utilities::MPI::ProcessGrid> process_grid,
```

Comment:
Why do you want to do this? (Also in the other patch you posted today?) The only thing that happens if you remove the reference is that you create a copy of the `shared_ptr` object, which has to subscribe to the object pointed to, and this copy then becomes the local function argument. At the end of the function, this function argument is destroyed and has to unsubscribe from the object pointed to.

That's two unnecessary subscription/unsubscription operations that can be avoided if you leave the argument here as a reference.

Reply:
I did it only for consistency. So far we had some functions/constructors passing a shared_ptr by value and some by const. refs. to the shared_ptr.

Reply:
Following your recommendation I changed the passing to const refs. to shared_ptr.